### PR TITLE
Clean code in codemeta_readme.R

### PR DESCRIPTION
**Pull request by hsonne**
Please have a look at my modifications in codemeta_readme.R. Again, my main goal was to remove duplication, to improve readability and to use consistent formatting. Thanks for reviewing!

### CODE REVIEW

File: R/codemeta_readme.R [hunk 1]
```diff
@@ -1,153 +1,164 @@
...
+  reviews <- jsonlite::read_json(url_onboarded_json)
 
-.ropensci_reviews <- function(){
-  reviews <-  jsonlite::read_json("https://badges.ropensci.org/json/onboarded.json")
-  reviews <- tibble::tibble(review = purrr::map_dbl(reviews, "iss_no"),
-                            package = purrr::map_chr(reviews, get_pkg_name))
+  reviews <- tibble::tibble(
+    review = purrr::map_dbl(reviews, "iss_no"),
+    package = purrr::map_chr(reviews, get_pkg_name)
+  )
```
Comment (maelle) [inline R/codemeta_readme.R]: is something returned?

File: R/codemeta_readme.R [hunk 2]
```diff
@@ -1,153 +1,164 @@
...
+  badges <- get_badge_links_matching(readme, url)
+
+  if (is.null(badges)) {
+
+    return(NULL)
   }
 
+  review <- as.numeric(stringr::str_remove(badges, paste0(".*https://", url)))
+
+  # TODO: What to do with more than one review or is that not possible?
```
Comment (maelle) [inline R/codemeta_readme.R]: more than one review from rOpenSci? not possible at the moment

File: R/codemeta_readme.R [hunk 3]
```diff
@@ -1,153 +1,164 @@
...
+codemeta_readme <- function(readme, codemeta) {
 
+  codemeta %>%
+    set_element_if_null("contIntegration", guess_ci(readme)) %>%
+    set_element_if_null("developmentStatus", guess_devStatus(readme)) %>%
+    set_element_if_null("review", guess_ropensci_review(readme))
+}
 
+# set_element_if_null ----------------------------------------------------------
+set_element_if_null <- function(x, element, value) {
```
Comment (maelle) [inline R/codemeta_readme.R]: could this go to utils.R?

Comment (hsonne) [reply R/codemeta_readme.R]: Of course, I will do that.

File: R/codemeta_readme.R (hunk 2 above)
Comment (hsonne) [reply R/codemeta_readme.R]: Ok, then I will remove the comment.

File: R/codemeta_readme.R (hunk 1 above)
Comment (hsonne) [reply R/codemeta_readme.R]: I assume that reviews is to be returned. I will remove the assignment.

Comment (maelle) [reply R/codemeta_readme.R]: 👌

File: R/codemeta_readme.R (hunk 2 above)
Comment (maelle) [reply R/codemeta_readme.R]: But good for me to keep in mind if we re-review packages in the future 😺

**Comment by maelle** [comment]
One of the tests is failing. 👀

File: R/codemeta_readme.R [hunk 4]
```diff
@@ -1,153 +1,148 @@
...
   }
 
-  if (is.null(codemeta$developmentStatus)){
-    codemeta$developmentStatus <-
-      guess_devStatus(readme)
+  readme_rmd <- grep("\\.[Rr]md$", readmes, value = TRUE)
+
+  if (length(readme_rmd) == 0) {
+
+    readme_file <- grep("\\.md$", readmes, value = TRUE)
```
Comment (maelle) [inline R/codemeta_readme.R]: `readme_file` does not get a value if `length(readme_rmd) != 0` which I think is the reason why one test is failing.

Comment (hsonne) [reply R/codemeta_readme.R]: You are right, my fault, sorry. I am fixing this.

Comment (maelle) [reply R/codemeta_readme.R]: no problem and no rush :-)

**Comment by hsonne** [comment]
Unfortunately, I cannot run the tests locally. They fail and I think that this has nothing to do with my changes, but I am not sure...
```
> test_check("codemetar")
-- 1. Error: we can generate codemeta
                    from the root directory of R source code on git
'codemetar_copy' exists and is not an empty directory
```

File: R/codemeta_readme.R [hunk 5]
```diff
@@ -107,23 +107,22 @@ guess_readme_path <- function(root) {
     return(NULL)
   }
 
+  # Filter for Rmd files
   readme_rmd <- grep("\\.[Rr]md$", readmes, value = TRUE)
 
-  if (length(readme_rmd) == 0) {
+  # If there is a Rmd file, use this one, else filter for md files
+  readme_file <- if (length(readme_rmd)) {
```
Comment (maelle) [inline R/codemeta_readme.R]: could you put the assignment inside the if and else? I personally find it easier to read :-)

**Comment by maelle** [comment]
Yes I think the test are now fine, I just have a stylistic comment and then will merge!

**Pull request merged**